### ISSUE TITLE
Overlay fails on empty polygon

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1260]
Overlay fails if there is `POLYGON EMPTY`. Raises RtreeError, which makes sense, but I would say that empty polygons should be rather ignored.

```py
from shapely.geometry import Point
from shapely import wkt
from geopandas import datasets, GeoDataFrame, read_file
from geopandas.tools import overlay

# NYC Boros
zippath = datasets.get_path('nybb')
polydf = read_file(zippath)

# Generate some circles
b = [int(x) for x in polydf.total_bounds]
N = 10
polydf2 = GeoDataFrame([
    {'geometry': Point(x, y).buffer(10000), 'value1': x + y, 'value2': x - y}
    for x, y in zip(range(b[0], b[2], int((b[2] - b[0]) / N)),
                    range(b[1], b[3], int((b[3] - b[1]) / N)))])

polydf2 = polydf2.append({'geometry': wkt.loads('POLYGON EMPTY'), 'value1': 2, 'value2': 3}, ignore_index=True)

newdf = overlay(polydf2, polydf)
```

<details>

```
RTreeError                                Traceback (most recent call last)
<ipython-input-68-30b6ee73f169> in <module>
      1 from geopandas.tools import overlay
----> 2 newdf = overlay(polydf2, polydf)

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/geopandas/tools/overlay.py in overlay(df1, df2, how, make_valid, use_sindex)
    389         return _overlay_difference(df1, df2)
    390     elif how == "intersection":
--> 391         result = _overlay_intersection(df1, df2)
    392     elif how == "symmetric_difference":
    393         result = _overlay_symmetric_diff(df1, df2)

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/geopandas/tools/overlay.py in _overlay_intersection(df1, df2)
    214     spatial_index = df2.sindex
    215     bbox = df1.geometry.apply(lambda x: x.bounds)
--> 216     sidx = bbox.apply(lambda x: list(spatial_index.intersection(x)))
    217     # Create pairs of geometries in both dataframes to be intersected
    218     nei = []

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/pandas/core/series.py in apply(self, func, convert_dtype, args, **kwds)
   4043             else:
   4044                 values = self.astype(object).values
-> 4045                 mapped = lib.map_infer(values, f, convert=convert_dtype)
   4046 
   4047         if len(mapped) and isinstance(mapped[0], Series):

pandas/_libs/lib.pyx in pandas._libs.lib.map_infer()

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/geopandas/tools/overlay.py in <lambda>(x)
    214     spatial_index = df2.sindex
    215     bbox = df1.geometry.apply(lambda x: x.bounds)
--> 216     sidx = bbox.apply(lambda x: list(spatial_index.intersection(x)))
    217     # Create pairs of geometries in both dataframes to be intersected
    218     nei = []

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/rtree/index.py in intersection(self, coordinates, objects)
    452             return self._intersection_obj(coordinates, objects)
    453 
--> 454         p_mins, p_maxs = self.get_coordinate_pointers(coordinates)
    455 
    456         p_num_results = ctypes.c_uint64(0)

~/anaconda3/envs/momepy_guide/lib/python3.7/site-packages/rtree/index.py in get_coordinate_pointers(self, coordinates)
    310         if len(coordinates) != dimension * 2:
    311             raise core.RTreeError(
--> 312                 "Coordinates must be in the form "
    313                 "(minx, miny, maxx, maxy) or (x, y) for 2D indexes")
    314 

RTreeError: Coordinates must be in the form (minx, miny, maxx, maxy) or (x, y) for 2D indexes
```
</details>

I would add `df1 = df1.loc[~df1.geometry.is_empty]`  and the same for `df2` with some warning after this line https://github.com/geopandas/geopandas/blob/add5fe9139883fa54d14fa28426478619948349c/geopandas/tools/overlay.py#L386 Or we can filter them out and then put back in the end, but that does not make much sense.

**Comment by erikburt** [comment]
I am experiencing this issue as well.

```python
import geopandas as gpd
from shapely.geometry import Point, Polygon

some_polygon = {'geometry': [Polygon([(-5.509260350101057, 36.260244201586794), (-5.731064912610456, 36.266167159517529), (-5.832776397076094, 36.250849164869088) , (-5.829100078360468, 36.285161472881597), (-5.579110405697939, 36.29190139052691), (-5.182524246397088, 36.260244201586794), (-5.182524246397088, 35.741474782826323), (-5.509260350101057, 35.741474782826323), (-5.509260350101057, 36.260244201586794)])]}
some_polygon_gdf = gpd.GeoDataFrame(some_polygon, geometry='geometry')

empty_gdf = gpd.GeoDataFrame()
emptypyth_gdf['geometry'] = None

result = gpd.overlay(empty_gdf, some_polygon_gdf, how="intersection")
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/geopandas-0.7.0-py3.6.egg/geopandas/tools/overlay.py", line 217, in overlay
    type = df1.geom_type.iloc[0]
  File "/usr/local/lib/python3.6/site-packages/pandas-1.0.1-py3.6-linux-x86_64.egg/pandas/core/indexing.py", line 1767, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/usr/local/lib/python3.6/site-packages/pandas-1.0.1-py3.6-linux-x86_64.egg/pandas/core/indexing.py", line 2137, in _getitem_axis
    self._validate_integer(key, axis)
  File "/usr/local/lib/python3.6/site-packages/pandas-1.0.1-py3.6-linux-x86_64.egg/pandas/core/indexing.py", line 2062, in _validate_integer
    raise IndexError("single positional indexer is out-of-bounds")
IndexError: single positional indexer is out-of-bounds
```

**Comment by martinfleis** [comment]
Fixed by #1428
